src/steps/os/unix.rs

```diff
@@ -1,21 +1,37 @@
-use crate::error::Error;
+use crate::error::{Error, ErrorKind::*};
 use crate::executor::{CommandExt, RunType};
 use crate::terminal::print_separator;
 use crate::utils::{require, which};
 use directories::BaseDirs;
+use std::env;
 use std::path::Path;
 use std::process::Command;
 
+fn zplug_exists(base_dirs: &BaseDirs) -> bool {
+    env::var("ZPLUG_HOME")
```

Comment:
You're probably right on that—I was just writing some Haskell and thinking in terms of monoids. I'll make those changes but how would you feel about my writing up some utility module that makes working with a bunch of paths easier? Specifically, I'm thinking it would allow you to chain a bunch of strings and then take the first one that's a valid path or something.